### PR TITLE
change signal handlers using write syscall instead of printf.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: memcached.c [hunk 1]
```diff
@@ -4277,15 +4277,25 @@ static void remove_pidfile(const char *pid_file) {
 
 static void sig_handler(const int sig) {
     stop_main_loop = EXIT_NORMALLY;
-    printf("Signal handled: %s.\n", strsignal(sig));
+    char *strsig = strsignal(sig);
```
Comment (en4bz) [inline memcached.c:4280]: `strsignal` is not reentrant.

**Comment by dormando** [comment]
Thanks!

To be honest I'm not sure I intended to leave printf's in the signal handlers. the side of the code picking up the signal should probably just printf instead of all this workaround..

**Comment by dormando** [comment]
Looking again at the `switch (stop_main_loop)` code I think these two prints can just be removed.

Maybe add an `if (verbose)` for `EXIT_NORMALLY` and move the term print there.

**Comment by dormando** [comment]
better, thanks!

**Comment by dormando** [comment]
released in 1.6.18

**Pull request closed without merging**